src/main/java/com/team2357/lib/subsystems/LimelightSubsystem.java

```diff
@@ -235,10 +361,72 @@ public double getInchesFromTarget() {
 
     return distance;
   }
-  /*
-  @Override
-  public void periodic() {
-    SmartDashboard.putNumber("Y", getTY());
+
+  public Pose2d getLimelightPose2d() {
+    return botposeToPose2d(m_limelightPoseInfoSub.get());
   }
-  */
+
+  public Long getLastTargetID() {
+    return m_Tid.get();
+  }
+
+  public Pose2d getRedPose() {
+    return botposeToPose2d(m_botposeWpiRed.get());
+  }
+
+  public Pose2d getBluePose() {
+    return botposeToPose2d(m_botposeWpiBlue.get());
+  }
+
+  public Pose2d getCurrentAllianceLimelightPose() {
+    if (DriverStation.getAlliance() == Alliance.Blue) {
+      return getBluePose();
+    } else if (DriverStation.getAlliance() == Alliance.Red) {
+      return getRedPose();
+    } else {
+      return null;
+    }
+  }
+
+  public double getBlueBotposeTimestamp() {
+    return calculateTimestamp(m_botposeWpiBlue.get());
+  }
+
+  public double getRedBotposeTimestamp() {
+    return calculateTimestamp(m_botposeWpiRed.get());
+  }
+
+  public double getCurrentAllianceBotposeTimestamp() {
+    if (DriverStation.getAlliance() == Alliance.Blue) {
+      return getBlueBotposeTimestamp();
+    } else if (DriverStation.getAlliance() == Alliance.Red) {
+      return getRedBotposeTimestamp();
+    } else {
+      return Double.NaN;
+    }
+  }
+
+  public double calculateTimestamp(double[] botpose) {
```

Comment:
This is a little unclear as to what it's doing. Is this the timestamp difference?